packages/core/src/modules/oob/messages/OutOfBandInvitation.ts

```diff
@@ -100,13 +100,26 @@ export class OutOfBandInvitation extends AgentMessage {
 
   // TODO: this only takes into account inline didcomm services, won't work for public dids
   public getRecipientKeys(): Key[] {
-    return this.services
-      .filter((s): s is OutOfBandDidCommService => typeof s !== 'string' && !(s instanceof String))
+    return this.getInlineServices()
       .map((s) => s.recipientKeys)
       .reduce((acc, curr) => [...acc, ...curr], [])
       .map((didKey) => DidKey.fromDid(didKey).key)
   }
```

Comment:
We should probably remove this method as it only takes into account inline didcomm services.

Reply:
It is still used on a few places. I'm not sure if it makes sense to remove it at this moment. What would be the replacement for the few spots, where it is used?

Reply:
Where is it used? The reason I'd like to remove it is that we should be taking into account public dids everywhere now. This methods makes it quite easy to forget the public did recipient keys, which can cause issues later

Reply:
Maybe it's better if you check for yourself:
1. `DidExchangeProtocol.processResponse`: https://github.com/hyperledger/aries-framework-javascript/blob/main/packages/core/src/modules/connections/DidExchangeProtocol.ts#L303 (not sure how to deal with this one)
2. `OutOfBandRecord` initial tags in constructor: Here we store the recipient keys from the inline services (this we could probably refactor by expanding the function in place)
3. `migrateToOobRecord`: https://github.com/hyperledger/aries-framework-javascript/blob/main/packages/core/src/storage/migration/updates/0.1-0.2/connection.ts#L315 (here it looks like we have to deal with inline services only as well, so probably easy to expand)

Reply:
Ok so I think we can remove / refactor them all:

1. This checks if the response is 'authorized' by the keys from the invitation. I think we can use the recipientKeyFingerprints from the tag for this and transform it into keys (`recipientKeysFingerprints.map(fingerprint => Key.fromFingerprint(fingerprint).publicKeyBase58)` )
2. I think we can remove the default tags assignment in the constructor. We should just set in once with the processing of both the dids and the inline services (it's redundant to set it in the constructor I think?)
3. You're right we only have to deal with inline services here. Maybe we can copy the implementation directly to the migration file.

Reply:
I have removed the `getRecipientKeys` method and refactored the 3 cases